css-variables/readme.md

```diff
@@ -0,0 +1,443 @@
+
+[Codepen Example](https://codepen.io/dsmmcken/pen/ZEPjBmy)
+
+```js
+document.addEventListener("mousemove", (e) => {
+  document.documentElement.style.setProperty("--x", e.clientX);
+  document.documentElement.style.setProperty("--y", e.clientY);
+});
+```
+
+```scss
+div {
+  transform: translate(var(--x) * 1px, var(--y) * 1px);
+}
+```
+
+
+
+
+
+
+
+
+## Start using css variables!
+
+We have a theme component which manages which theme is applied. All users created themes must inheriet from either a light or a dark theme.
+
+1. All Deephaven **theme variables** start with `--dh` to avoid conflicts with other css variables. `--dh` variables should be thought of as a public API. Removing a `--dh` variable is a breaking change. Adding a variable is not a breaking change, as users themes inherit from the base themes, and will not break if a variable is added.
+ 
+   Local variables to components should not start with `--dh`, but instead should follow the component name `--my-component`.
+
+2. A theme consists of a set of css variables defined across a number of files. Variables added to Dark or Light theme **must be added to both base theme files**. You will rarely need to add a variable to a theme file unless we think it should be part of the public api.
+
+3. Components should use the most specific available variable. If a **semantic _component_ variable** is available use that, otherwise it should use a **semantic variable** and lastly it should use a **palette variable**.
+
+4. Components should almost never use palette variables directly.
+
+   <small>Sometimes maybe a gray-*, but really mostly no.</small>
+
+5. Components can and should define their own internal variables, scoped to the component.
+
+
+Please review the /styleguide for a complete list of exposed variables.
+
+
+
+
+### Stop using SCSS color variables!
+
+Immediately discontinue use of scss color variables in favor of css variables.
+
+- scss unit variables like $spacer-*, or $border-radius are still ok to use (for now).
+- All bootstrap variables have been re-defined to use the css variables, but we haven't removed the scss variables yet. They should be considered deprecated.
+  ex. 
+  ```scss
+  $content-bg: var(--dh-color-content-bg);
+  $primary: var(--dh-color-accent);
+  ```
+- Immediately **stop using compile time color manipulation functions in scss**. This is no longer possible darken($primary, 10%) does not work against a runtime variable. However, good news is with the expanded palette, this is no longer necessary. For ex. Accent 100-1400 covers a much wider range of colors than we had before. All existing scss color manipulations were removed and replaced as part of themeing work.
+- Transparency is a special case. Since user supplied variables can be in any sRGB formart we must also stop using scss functions for transparency. Instead of `rgba($color, 0.5)` use either a color-mix function:
```

Comment:
```suggestion
- Transparency is a special case. Since user supplied variables can be in any sRGB format we must also stop using SCSS functions for transparency. Instead of `rgba($color, 0.5)` use either a color-mix function:
```